Introduce a new multi-select cell type.

jansiegel opened the pull request:
### Context
This PR introduces a new **MultiSelect cell type** (editor, renderer, and validator).

The MultiSelect cell type allows storing multiple values (or key/value pairs) in a single cell.

#### Overview

- ##### Editor
    The editor is a dropdown with selectable items.  
    Items can be selected by clicking them or by using the keyboard:
    - navigate with arrow keys,
    - select using `SPACE` or `ENTER` (configurable via an option).

    Each selection immediately updates the underlying cell data (unlike the other Handsontable editors).

- ##### Data format
    The cell value is always an array:
    - an array of strings, or
    - an array of objects with `key` / `value` properties,  
    depending on how the `source` property is defined in the cell meta.

- ##### Renderer
    The renderer displays the selected values as a list of closable tags (“chips”), reflecting the underlying cell data.

- ##### Validator
    The validator verifies that:
    - the cell value matches the expected data type,
    - the stored values are a subset of the values defined in the cell’s `source`.

#### Available options
*(Option names subject to change; discussion required)*

- **`allowEmpty`** – allows empty values to pass validation (default: `true`)
- **`placeholder`** – placeholder text shown in the cell renderer when the cell is empty
- **`visibleRows`** – maximum number of visible rows in the dropdown
- **`maxSelections`** – maximum number of selectable entries
- **`sourceSortFunction`** – custom sort function for dropdown entries
- **`enterCommits`** – defines whether pressing `ENTER` closes the editor
- **`searchInput`** – toggles the search input inside the dropdown
- **`filteringCaseSensitive`** – if `true`, search filtering is case sensitive

#### Still left to do

1. **CSS variables**  
   New UI elements require new CSS variables. Temporary variables are currently used, as agreed with @sl01k.

2. **Performance and refactoring (Phase 2)**  
   There are potential performance improvements to be addressed (e.g., the dropdown may re-render more often than necessary). No visible issues were observed during regular usage.  
   Unifying the autocomplete, dropdown, and select editors will likely require a small refactor.

3. **Accessibility**  
   ARIA attributes will be updated after Wojtek’s review.

#### Demo:
- https://j27yll.csb.app/ (based on `0.0.0-next-a4c3bb2-20260211`)

-------

| :exclamation:  Discrepancies between the final implementation and the original specification are documented in this gist: https://gist.github.com/jansiegel/be52e96243da30743ebda26f03eb69b0 .   |
|----------------------------------------------|

> [!IMPORTANT]  
> **NOTE: The legacy theme tests are failing, because (as instructed) I did not prepare the styles for the legacy theme.**

-------

### How has this been tested?
Added new test cases for the editor, renderer and validator.

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. handsontable/dev-handsontable#2958
2. https://app.clickup.com/9015210959/v/dc/8cnjcyf-6575

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/angular-wrapper`
- [ ] `@handsontable/react`
- [ ] `@handsontable/react-wrapper`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

jansiegel committed:
[WIP] Initial MultiSelect cell type implementation.

jansiegel committed:
[WIP] Add support for handling key/value objects (non-objects require additional work)

jansiegel committed:
[WIP] Implement back support for non-key/value sources.

jansiegel committed:
[WIP] Implement the vertical flipping mechanism + fixing support for key/value and other changes.

jansiegel committed:
[WIP] Implement new features, refactor, fixing lint errors.

jansiegel committed:
[WIP] Add test cases for the current state of MultiSelect editor + fix problems found while writing test cases.

jansiegel committed:
[WIP] Fix a problem with autofilled cell format for newly created rows for multiSelectEditor and autofillEditor (with key/value source) + add tests.

jansiegel committed:
[WIP]
- Implement tests for the keyboard navigation
- Add the maxSelections option

jansiegel committed:
Fix the lint errors.

jansiegel committed:
[WIP] Add the `caseSensitiveFiltering` option + tests.

jansiegel committed:
[WIP] add a source sorting option + tests

jansiegel committed:
[WIP] Implement additional options + tests.

jansiegel committed:
Merge remote-tracking branch 'origin/develop' into feature/dev-issue-2958

jansiegel committed:
[WIP] Fix the unit tests.

jansiegel committed:
[WIP] Fix lint errors.

jansiegel committed:
[WIP] Replace the regex logic with a simpler one in one of the helpers.

jansiegel committed:
[WIP] Clean up unused properties and make some of them private.

jansiegel committed:
[WIP] Make the entire list element clickable + modify the autofill complex data logic.

jansiegel committed:
[WIP] Correct lint errors + cleanup.

jansiegel committed:
[WIP] Dropdown sizing changes.

jansiegel committed:
[WIP] Ditch the two-way input editor in favor of a chips renderer.

jansiegel committed:
[WIP] Further advancements in the chips renderer development.

jansiegel committed:
[WIP] Fix synchronizing editor position on scroll.

jansiegel committed:
[WIP] Clean up the code and fix minor issues.

jansiegel committed:
[WIP] Fix lint errors.

jansiegel committed:
[WIP] Fix unit tests.

jansiegel committed:
[WIP] Correct the tests + change the dropdown items' id naming.

jansiegel committed:
[WIP] Correct lint errors.

jansiegel committed:
[WIP] Modify test css environment.

jansiegel committed:
Post-pre-review changes.

jansiegel committed:
Fix lint errors.

jansiegel committed:
Update the dropdown height logic.

jansiegel committed:
- Add the missing new options to metaSchema
- Correct the test cases (make them use CSS var values)
- Style the chips renderer and dropdown accordingly with the specification

jansiegel committed:
- Clean up + add tests.
- Correct the dropdown's visible row calculation.

jansiegel committed:
[tmp] debug test

jansiegel committed:
[tmp] debug test

jansiegel committed:
[tmp] debug test

jansiegel committed:
[tmp] debug test

jansiegel committed:
[tmp] debug test

jansiegel committed:
Fix a failing test for the main theme.

jansiegel committed:
Add a new test case.

jansiegel committed:
- Implement a validator for the multiSelect type
- Add tests for the validator
- Fix a bug with copy/paste
- Refactor the code

jansiegel committed:
Add the changelog entry.

jansiegel committed:
Fix the unit tests.

jansiegel commented:
@Wojciech-Swiderski could you check for the missing ARIA tags (and any other accessibility issues)?

budnix commented on handsontable/src/3rdparty/walkontable/src/event.js:
```diff
@@ -188,6 +188,7 @@ class Event {
     if (!['TD', 'TH'].includes(activeElement.nodeName) &&
       (
         realTarget === activeElement ||
+        hasClass(realTarget, EDITOR_CONTROLLER_CLASS) ||
```
This seems to introduce hard-coded behavior for the editor. Would it be possible to refactor this into a more generic solution?

budnix commented on handsontable/src/dataMap/metaManager/metaSchema.js:
```diff
@@ -2265,6 +2265,29 @@ export default () => {
+    /**
+     * The `enterCommits` option configures whether the <kbd>**Enter**</kbd> key closes the [`multiSelect`](@/guides/cell-types/multiselect-cell-type/multiselect-cell-type.md) editor.
+     *
+     * @memberof Options#
```
Missing `@since` tag here and there. Also, worth adding that this option works only for `multiSelect` cell type (at least now).

budnix commented on handsontable/src/editors/multiSelectEditor/controllers/dropdownController/dropdownController.js:
```diff
@@ -0,0 +1,580 @@
+import { mixin } from '../../../../helpers/object';
```
If possible, I'd flatten the directory structure. Seems still very logical, but there is no burden while navigating files. WDYT?

<img width="358" height="318" alt="image" src="https://github.com/user-attachments/assets/e6556116-7ae8-4739-9470-31caeb6c6eca" />

budnix reviewed (changes requested):
When `source` is not provided there is an error on cell click
```
array.js:297 Uncaught TypeError: Cannot read properties of undefined (reading 'some')
    at array.js:297:1
    at array.js:304:1
    at Array.filter (<anonymous>)
    at array.js:304:1
    at arrayEach (array.js:180:1)
    at getIntersectionOfArrays (array.js:303:1)
    at getValuesIntersection (utils.js:33:1)
    at MultiSelectEditor.prepare (multiSelectEditor.js:117:1)
    at EditorManager.prepareEditor (editorManager.js:132:1)
    at Selection.<anonymous> (core.js:598:1)
 ```

jansiegel committed:
- Post CR change (more generic chip-closing logic)
- Fix a problem with wrong dropdown position calculation when the dropdown was opened upwards.

jansiegel committed:
Post-CR changes (reorganizing the editor file structure + handling a not provided 'source' property)

jansiegel committed:
Fix lint errors.

jansiegel committed:
- Rename the cell type "multiSelect" to "multiselect" to retain the consistency with the other cell types, like "autocomplete"
- Fix a bug with the search bar icon styles being overriden because of a vague selector
- Fix a bug with removing entries using the chip-closing buttons did not enable the dropdown checkboxes with the maxSelections option enabled
- Add a documentation page and demos for the new cell type

jansiegel committed:
Correct a leftover mention of a temporary css variable.

jansiegel committed:
Merge remote-tracking branch 'origin/develop' into feature/dev-issue-2958

jansiegel committed:
Rename "multiSelect" to "multiselect" in the validator, editor and renderer.

GreenFlux reviewed (dismissed):
New multi-select docs look good. The PR mentions keyboard navigation, but the docs do not. May consider adding a note on keyboard bindings.

jansiegel committed:
Extended the docs page with information on keyboard navigation.

jansiegel commented:
> New multi-select docs look good. The PR mentions keyboard navigation, but the docs do not. May consider adding a note on keyboard bindings.

Good point - added in https://github.com/handsontable/handsontable/pull/11981/commits/da28b80c72374fa66135c2f8fdac42e2a10b161c 👍 .

jansiegel commented:
This PR can be merged only after https://github.com/handsontable/handsontable/pull/11950 is done - some of the changes of the styles in this PR need to be verified after Theme API PR is done.

Wojciech-Swiderski commented:
When I navigate via keyboard I can open slect in two ways - on enter and on space but when opened on space I cannot select ceckboxes with arrows

jansiegel commented:
> When I navigate via keyboard I can open slect in two ways - on enter and on space but when opened on space I cannot select ceckboxes with arrows

@Wojciech-Swiderski You're right. Opening using anything other than <kbd>ENTER</kbd> was not planned, tbh. It comes from the "quick edit" feature of the text editor.
Do you think it should be supported, or should I block opening the editor with anything other than <kbd>ENTER</kbd>?

Wojciech-Swiderski commented:
<img width="429" height="279" alt="image" src="https://github.com/user-attachments/assets/d3a0e7af-8b1b-4d94-859c-6ac1656b9278" />
When in search I write something I expect to see only maching options. Right now I see maching unchecked options and checked options. 

I believe that only matches should be shown

Wojciech-Swiderski commented:
> > When I navigate via keyboard I can open slect in two ways - on enter and on space but when opened on space I cannot select ceckboxes with arrows
> 
> @Wojciech-Swiderski You're right. Opening using anything other than ENTER was not planned, tbh. It comes from the "quick edit" feature of the text editor. Do you think it should be supported, or should I block opening the editor with anything other than ENTER?

I do believe we should support it

jansiegel commented:
> <img alt="image" width="429" height="279" src="https://private-user-images.githubusercontent.com/197734205/547096687-d3a0e7af-8b1b-4d94-859c-6ac1656b9278.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2NDE3MDAsIm5iZiI6MTc3MDY0MTQwMCwicGF0aCI6Ii8xOTc3MzQyMDUvNTQ3MDk2Njg3LWQzYTBlN2FmLThiMWItNGQ5NC04NTljLTZhYzE2NTZiOTI3OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIwOVQxMjUwMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDIwNWRlYzM0Mjg2OWYwYzdlNTRiZDQ4YTE3Y2RlZDU5YzJjZGNlNDBlNjliOGFiNWM1YmFkOTIyNTkzNTdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.N2vjID3XsLzC7dJhOhLfcld0NGkXYpATDS2n0gsG35I"> When in search I write something I expect to see only maching options. Right now I see maching unchecked options and checked options.
> I believe that only matches should be shown

-----

@Wojciech-Swiderski This functionality comes from this comment: [[clickup link](https://app.clickup.com/9015210959/v/dc/8cnjcyf-6575/8cnjcyf-7595?comment=90150176243582&threadedComment=90150176610643)].
>  I see one issue, though: selected items should remain visible. Right now they disappear, which may cause confusion.

This was specified for the pre-chips editor/renderer version of the MultiSelect cell type, though. 
@sl01k, could you confirm how should the filtering work?

Wojciech-Swiderski commented:
https://github.com/user-attachments/assets/f714da40-3fea-45ab-9eb8-b68d858c0299

When I 

1. Choose a cell and press enter
2. Choose an option and press space
3. Press esc key  
4. then press enter again 
there is an empty dropdown

Wojciech-Swiderski commented:
I spoted cuple of accessibility issues:

- Search should have aria-label, placeholder isn't enough for screen readers. aria-label="Search options" would be the best option here I believe
- div with magnyfying glass in search box should have aria-hidden added to it
- Whole dropdown should have role and aria-label. I believe role="group" would be the best with aria-label="Select options"

Wojciech-Swiderski commented:
When maxSelections is checked and user navigates via keyboard:

1. When selections are maxed out, and user moves focus to an element that is not checked, focus indicator isn't visible on that element

2. When selections are maxed out, focus is on Search box and first element isnot checked user cannot use arrows to reach checkboxes

https://github.com/user-attachments/assets/efb7948b-1e03-45b0-867a-6d929deea5c2

Wojciech-Swiderski commented:
When maxSelections is checked and user navigates via keyboard:

- When I max out number of selections
- move with tab to next column (without pressing enter nor esc)
- go back and press enter and navigate to an option that is unchecked - I can check this option

https://github.com/user-attachments/assets/2dafd2c2-532f-411b-82ea-01147a71b1ed

Wojciech-Swiderski commented:
When 
- in second column multiselect is opened
- user clicks "x" in the first column and clicks again in empty space in the same cell
cell in second column is filled by chips from first column

https://github.com/user-attachments/assets/9df73ae2-738e-42ef-87c5-4651e3297248

jansiegel committed:
Fix some of the problems reported in the code review.

Wojciech-Swiderski commented:
On Safari when something is written in search, focus is shown on the side of the input: 
<img width="262" height="188" alt="image" src="https://github.com/user-attachments/assets/8ef0a28b-9012-425e-ad86-864102da15ee" />

on chrome:
<img width="125" height="59" alt="image" src="https://github.com/user-attachments/assets/26ce4b52-b70d-4aac-a2d9-81643a0173b6" />

jansiegel committed:
Make keeping the selected items during filtering optional and configurable via a setting.

jansiegel committed:
Add the missing aria tags.

jansiegel committed:
Replace the "disabled" state of the checkboxes with a data- attribute to allow focusing on the entries using the keyboard.

jansiegel committed:
Fix a bug, where removing a chip from another column affected the selection of the currently edited column.

jansiegel committed:
Merge remote-tracking branch 'origin/develop' into feature/dev-issue-2958

Wojciech-Swiderski commented:
I have question about two things:
1. When text is autofilled from multiselect to text cell it shows name of chip but when it is autoffiled back (text to multicell) it shows that chip is incorrect even if chip with said name exists should it behave like this?

<img width="589" height="227" alt="image" src="https://github.com/user-attachments/assets/d49cb91c-efa1-4495-924e-593394ae5739" />

2. when multiple chips are autofilled to text it shows values separated by comma, when it is autofilled back shouldn't it autofill into 3 separate chips?

https://github.com/user-attachments/assets/7d2fd214-3419-4492-bfbf-5f9afab92c4f

3. When there is too much text in chip, we get elipsis andtext get truncated. When user sees truncated text they cannot see whole text without entering editor. Shouldn't we add title attribute to show whole text when user hovers over it? or add marquee like text on hover?

jansiegel committed:
Sync with the theme api changes, add the icons using the handsontable-figma script.

jansiegel commented:
> I have question about two things:
> 
> 1. When text is autofilled from multiselect to text cell it shows name of chip but when it is autoffiled back (text to multicell) it shows that chip is incorrect even if chip with said name exists should it behave like this?
> 
> <img alt="image" width="589" height="227" src="https://private-user-images.githubusercontent.com/197734205/547674205-d49cb91c-efa1-4495-924e-593394ae5739.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA3MzQ2MjYsIm5iZiI6MTc3MDczNDMyNiwicGF0aCI6Ii8xOTc3MzQyMDUvNTQ3Njc0MjA1LWQ0OWNiOTFjLWVmYTEtNDQ5NS05MjRlLTU5MzM5NGFlNTczOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIxMFQxNDM4NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NTIxN2Y5ZjZlYmNhY2UyZWQ0NDYyZGE4MGUwMjM2ZjlmMDMzMDE0MTYwMWY2NWRhODEyZWMxMGJjYzFkMDIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4nEZVkZNiy4J84dxbclDfqHfB5Vr4hzwvQkBBkUDZ9E">
> 2. when multiple chips are autofilled to text it shows values separated by comma, when it is autofilled back shouldn't it autofill into 3 separate chips?
> 
>  Screen.Recording.2026-02-10.at.12.48.17.mov 
> 3. When there is too much text in chip, we get elipsis andtext get truncated. When user sees truncated text they cannot see whole text without entering editor. Shouldn't we add title attribute to show whole text when user hovers over it? or add marquee like text on hover?

@Wojciech-Swiderski 
### As for **1** and **2**:
That needs to be handled somehow, but I don't think it's described in the specification.

The problem is:
1. The "regular" (text) cells hold strings as values.
2. The `multiselect`-typed cells hold arrays as values.
3. When we autofill/copy+paste a value from a `multiselect`-typed cell to a "regular" (text) cell, the array gets flattened and strigified with a comma as a delimiter.
4. When we autofill/copy+paste a value from a "regular" cell _back_ to a `multispect`-typed cell, that comma-separated string does not get parsed to an array. The value is just duplicated, leaving the `multiselect`-typed cell with an invalid, non-array value (that's why it doesn't pass validation)

edit: There's also logic that tries to parse a string into an array, if the string is a JSON array (needed for copy/paste).

My intuition tells me, that in this situation we should not try to render the invalid value as a chip, but display an "#bad-value#" message from the renderer (similarly to the checkbox renderer). 
@sl01k Could you help us here?

### As fro **3**:
I think it's a good idea. If @sl01k doesn't oppose it, I'll add the `title` to the chips.

jansiegel committed:
- Add the #bad-value# functionality
- Add title attributes to the chips renderer

jansiegel commented:
@Wojciech-Swiderski @sl01k, I updated the demo with today's build (added `#bad-value#` and `title`): https://thg32r.csb.app/

sl01k commented:
> @Wojciech-Swiderski
> 
> ### As for **1** and **2**:
> That needs to be handled somehow, but I don't think it's described in the specification.
> 
> The problem is:
> 
> 1. The "regular" (text) cells hold strings as values.
> 2. The `multiselect`-typed cells hold arrays as values.
> 3. When we autofill/copy+paste a value from a `multiselect`-typed cell to a "regular" (text) cell, the array gets flattened and strigified with a comma as a delimiter.
> 4. When we autofill/copy+paste a value from a "regular" cell _back_ to a `multispect`-typed cell, that comma-separated string does not get parsed to an array. The value is just duplicated, leaving the `multiselect`-typed cell with an invalid, non-array value (that's why it doesn't pass validation)
> 
> edit: There's also logic that tries to parse a string into an array, if the string is a JSON array (needed for copy/paste).
> 
> My intuition tells me, that in this situation we should not try to render the invalid value as a chip, but display an "#bad-value#" message from the renderer (similarly to the checkbox renderer). @sl01k Could you help us here?
> 
> ### As fro **3**:
> I think it's a good idea. If @sl01k doesn't oppose it, I'll add the `title` to the chips.

Ad3 - Agree attr title is good idea

Ad1, Ad2 – at this point, we have the same behaviour as in the case of a drop-down list with a key/value – we should have a common solution for this type of situation; if we accepted it earlier for key/value, then it is also acceptable now. 
Ag-grid also did not solve this problem: https://codesandbox.io/p/sandbox/value-handlers-forked-3s6m86?file=%2Fmain.ts%3A35%2C23

jansiegel committed:
Reverse the "#bad-value#" changes.

jansiegel committed:
Fix a lint error.

Wojciech-Swiderski commented:
> > When I navigate via keyboard I can open slect in two ways - on enter and on space but when opened on space I cannot select ceckboxes with arrows
> 
> @Wojciech-Swiderski You're right. Opening using anything other than ENTER was not planned, tbh. It comes from the "quick edit" feature of the text editor. Do you think it should be supported, or should I block opening the editor with anything other than ENTER?

@jansiegel I still see the issue when the search is hidden, I can open it on space but i cannot use arrows inside dropdown

https://github.com/user-attachments/assets/5287dbf6-f086-41fb-9440-0dc070e040d7

Wojciech-Swiderski commented:
@sl01k 
Another question apeared in my mid when re-checking changes
When:
- Search is on
- Visible rows are on

and we search non-existent chip, shouldn't there be moreempty space - like for 4 elements in the way that it is in column menu

<img width="992" height="510" alt="image" src="https://github.com/user-attachments/assets/10740918-cb2e-4fad-97de-3144ed3a31a4" />

<img width="279" height="294" alt="image" src="https://github.com/user-attachments/assets/122cf57c-23a9-4a4f-9337-c97c138c795d" />

jansiegel committed:
Fix a problem with opening a searchless dropdown with SPACE.

jansiegel commented:
> @jansiegel I still see the issue when the search is hidden, I can open it on space but i cannot use arrows inside dropdown

@Wojciech-Swiderski You're right - should be good after https://github.com/handsontable/handsontable/pull/11981/commits/a4c3bb297ceb5d6aefddc1edea899cbcac6ea9ef.

Updated demo: https://j27yll.csb.app/

Wojciech-Swiderski reviewed (dismissed):
Tested

budnix commented on handsontable/src/styles/components/renderers/_multi-select-renderer.scss:
```diff
@@ -0,0 +1,85 @@
+@use "../../utils/mixins";
```
About scss file maybe it'd be better to reneme the file to `_multi-select-renderer.scss‎` for coherence. WDYT?

jansiegel replied:
Good idea 👍 Done in https://github.com/handsontable/handsontable/pull/11981/commits/db3d75ec34bfa5126489c1d106498af5fa5c66d7

jansiegel committed:
Rename the multiselect scss files.

jansiegel committed:
Correct import paths.

jansiegel committed:
Correct import paths.

ajit-appventurez commented:
Default Filter is not working when cell is multi-select
